frontend/src/metabase/metadata/pages/DataModel/components/TableSection/FieldList/FieldItem/FieldItem.tsx

```diff
@@ -0,0 +1,161 @@
+  const [updateField] = useUpdateFieldMutation();
+  const [sendToast] = useToast();
+  const icon = getColumnIcon(Lib.legacyColumnTypeInfo(field));
+
+  const handleNameChange = async (name: string) => {
+    if (field.display_name === name) {
+      return;
+    }
+
+    const { error } = await updateField({ id, display_name: name });
+
+    if (!error) {
+      sendToast({
+        icon: "check",
+        message: t`Display name for ${name} updated`,
+      });
+    }
+  };
+
+  const handleDescriptionChange = async (event: {
+    target: HTMLTextAreaElement;
+  }) => {
+    const description = event.target.value;
+    const newDescription = description.trim();
+
+    if ((field.description ?? "") === newDescription) {
+      return;
+    }
+
+    const { error } = await updateField({
+      id,
+      // API does not accept empty strings
+      description: newDescription.length === 0 ? null : newDescription,
+    });
+
+    if (!error) {
+      sendToast({
+        icon: "check",
+        message: t`Description for ${getFieldDisplayName(field)} updated`,
+      });
+    }
+  };
+
+  const handleInputClick = (event: MouseEvent) => {
+    // EditableText component breaks a11y with the programmatic
+    // event.currentTarget.click() call. The click event bubbles
+    // to this Link component. This is problematic e.g. when tabbing
+    // out of the EditableText component, as the link gets clicked.
+    if (!event.isTrusted) {
+      event.stopPropagation();
+      event.preventDefault();
+    }
+  };
+
+  const handleClick = (event: MouseEvent) => {
+    // prevent navigation when focusing textareas with mouse
+    if (
+      event.target instanceof HTMLElement &&
+      event.target.tagName === "TEXTAREA"
+    ) {
+      event.preventDefault();
+    }
+  };
+
+  return (
+    <Flex
+      aria-label={field.display_name}
+      bg={active ? "brand-lighter" : "bg-white"}
+      c="text-medium"
+      className={cx(S.field, {
+        [S.active]: active,
+      })}
+      component={Link}
+      direction="column"
+      align="flex-start"
+      gap={rem(12)}
+      justify="space-between"
+      mih={rem(40)}
+      pos="relative"
+      px="md"
+      py={rem(12)}
+      role="listitem"
+      to={href}
+      w="100%"
+      wrap="nowrap"
+      onClick={handleClick}
+    >
+      <Group flex="0 0 auto" gap="sm" maw="100%" wrap="nowrap">
+        <Icon className={S.icon} flex="0 0 auto" name={icon} />
+
+        <Box
+          className={S.name}
+          component={EditableText}
+          fw="bold"
+          initialValue={field.display_name}
+          maxLength={254}
+          miw={0}
+          mb={rem(-4)}
+          mt={rem(-3)}
+          ml={rem(-2)}
+          px={rem(1)}
+          py={rem(2)}
+          placeholder={t`Give this field a name`}
+          tabIndex={undefined} // override the default 0 which breaks a11y
+          onChange={handleNameChange}
+          onClick={handleInputClick}
+        />
```

Comment:
Because it's an input, my browser also marks typo's with red squiggly lines which might happen often for table names and it feels weird, as if there are errors with the table.

<img width="371" alt="Screenshot 2025-06-17 at 17 53 41" src="https://github.com/user-attachments/assets/258f5b46-160c-44da-873a-da8246cd9dfc" />